chore(repo): use shell-emulator for better cross-platform compatibility

danielbayley opened the pull request:
<!--
  ⚡️ katchow! We ❤️ Pull Requests!

  If you remove or skip this template, you'll make the 🐼 sad and the mighty god
  of Github will appear and pile-drive the close button from a great height
  while making animal noises.

  Pull Request Requirements:
  * Please include tests to illustrate the problem this PR resolves.
  * Please lint your changes by running `npm run lint` before creating a PR.
  * Please update the documentation in `/docs` where necessary

  Please place an x (no spaces - [x]) in all [ ] that apply.
-->

<!-- the plugin(s) this PR is for -->

## Rollup Plugin Name: `*`

This PR contains:

- [x] bugfix
- [ ] feature
- [x] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [ ] yes (_bugfixes and features will not be merged without tests_)
- [x] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

If yes, then include "BREAKING CHANGES:" in the first commit message body, followed by a description of what is breaking. 

List any relevant issue numbers:

<!--
If this PR resolves any issues, list them as

  resolves #1234

where 1234 is the issue number. This will help us with house-keeping as Github will automatically add a note to those issues stating that a potential fix exists. Once the PR is merged, Github will automatically close those issues.

If an issue is only solved partially or is relevant in some other way, just list the number without "resolves".
-->

### Description

<!--
  Please be thorough and clearly explain the problem being solved.
  * If this PR adds a feature, look for previous discussion on the feature by searching the issues first.
  * Is this PR related to an issue?
-->

I started work on a separate PR, but quickly ran into errors with the `prepare` `scripts` apparently not being cross-platform (`if` command not found—I am on macOS). Since this is a `pnpm` monorepo, and the `scripts` are very simple, why not make use of the [`shell-emulator`](https://pnpm.io/cli/run#shell-emulator) `.npmrc` config option (which uses [yarnpkg-shell](https://yarnpkg.com/features/scripting#portable-shell))? This also means the dependency on `del-cli` is no longer necessary.

tada5hi commented:
failing wasm test should be unrelated to pr?!
@shellscape

shellscape reviewed (approved):
@danielbayley I'm good to merge this once the conflicts are resolved. To resolve the pnpm-lock conflict, run `pnpm i` in the repo root and pnpm will take care of it for us.

danielbayley commented:
@shellscape Should be resolved now…

shellscape commented:
Closing as abandoned. Please feel free to reopen if you get back around to it.

danielbayley commented:
> Closing as abandoned. Please feel free to reopen if you get back around to it.

@shellscape I resolved the issue, as requested 🤷🏻‍♂️

danielbayley commented:
> > Closing as abandoned. Please feel free to reopen if you get back around to it.
> 
> 
> 
> I resolved the issue, as requested @shellscape 🤷🏻‍♂️
> 
> 
> 
> My other PR (#1744) is on hold only because/if #1782 is set to supercede it…
> 
> 
> 
> But as already mentioned, this one should be good to go! @shellscape

Bump @shellscape

shellscape commented:
@danielbayley ugh. that stupid corepack bug has us blocked in our workflows. this is going to take me a couple of days to get to.

danielbayley commented:
> @danielbayley this one is quite old, but we've fixed up the repo since. if you update from upstream/main I think we'll be in good shape

@shellscape Ok cool. Rebased…
